### PR TITLE
Amadeus: DSP code generation improvements

### PR DESCRIPTION
**Pull request by Thog**
This improve RyuJIT codegen drastically on the DSP side.
This may reduce CPU usage of the DSP thread quite a lot.

There is still room to improvements on that side and this might be addressed by follow up PRs.

As always, regression testing needed.

### CODE REVIEW

**Comment by GamerzHell9137** [comment]
Are there any places where we should look into where we could see differences?
Which audio backend etc?

File: Ryujinx.Audio/Renderer/Dsp/Command/CommandList.cs [hunk 1]
```diff
@@ -70,14 +81,47 @@ public void AddCommand(ICommand command)
             throw new NotImplementedException();
         }
 
-        public Memory<float> GetBufferMemory(int index)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public unsafe IntPtr GetBufferPointer(int index)
+        {
+            if (index > 0 && index < _buffersEntryCount)
```
Comment (Thealexbarney) [inline Ryujinx.Audio/Renderer/Dsp/Command/CommandList.cs:87]: index >= 0

**Review by Thealexbarney — CHANGES_REQUESTED** [review]
Looks mostly good. And I have to say you're quite aggressive with `AggressiveInlining`

File: Ryujinx.Audio/Renderer/Dsp/Command/DepopForMixBuffersCommand.cs [hunk 2]
```diff
@@ -57,10 +57,15 @@ public DepopForMixBuffersCommand(Memory<float> depopBuffer, uint bufferOffset, u
             }
         }
 
-        private float ProcessDepopMix(Span<float> buffer, float depopValue, uint sampleCount)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe float ProcessDepopMix(float* buffer, float depopValue, uint sampleCount)
         {
-            if (depopValue <= 0)
+            Debug.Assert(depopValue == 0);
```
[the remark this answers is not in the document]
Comment (Thog) [reply Ryujinx.Audio/Renderer/Dsp/Command/DepopForMixBuffersCommand.cs:63]: Technically can be removed as this should never be called with 0

File: Ryujinx.Audio/Renderer/Dsp/Command/DelayCommand.cs [hunk 3]
```diff
@@ -88,13 +90,14 @@ private void ProcessDelayMono(Span<float> outputBuffer, ReadOnlySpan<float> inpu
             }
         }
 
-        private void ProcessDelayStereo(Memory<float>[] outputBuffers, ReadOnlyMemory<float>[] inputBuffers, uint sampleCount)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```
[the remark this answers is not in the document]
Comment (Thog) [reply Ryujinx.Audio/Renderer/Dsp/Command/DelayCommand.cs:93]: I guess as I have all of those with known channel count I could make those constants here, thanks 👍

**Review by Thealexbarney — APPROVED** [review]
Looks okay to me

**Review by AcK77 — APPROVED** [review]

**Pull request merged**